src/Functions/hilbertEncode.cpp

```diff
@@ -0,0 +1,59 @@
+#include <Functions/hilbertEncode.h>
+#include <Functions/FunctionFactory.h>
+
+
+namespace DB
+{
+
+REGISTER_FUNCTION(HilbertEncode)
+{
+    factory.registerFunction<FunctionHilbertEncode>(FunctionDocumentation{
+        .description=R"(
+Calculates code for Hilbert Curve for a list of unsigned integers
+
+The function has two modes of operation:
+- Simple
+- Expanded
+
+Simple: accepts up to 2 unsigned integers as arguments and produces a UInt64 code.
+[example:simple]
```

Comment:
Change "SELECT hilbertEncode(1, 2, 3)" to "SELECT hilbertEncode(1, 2)" since the function is described to accept up to 2 integers. Explain that hilbertEncode(1, 2) computes the Hilbert index for the 2D point (1, 2).